rosbag2_transport/test/rosbag2_transport/test_record_options.cpp

```diff
@@ -42,7 +42,8 @@ TEST(record_options, test_yaml_serialization)
   std::stringstream serializer;
   serializer << node;
   auto reconstructed_node = YAML::Load(serializer.str());
-  auto reconstructed = reconstructed_node.as<rosbag2_transport::RecordOptions>();
+  auto reconstructed = YAML::decode_for_version<rosbag2_transport::RecordOptions>(
+    reconstructed_node, 9);
```

Comment:
Curious if we can make the original `reconstructed_node.as<rosbag2_transport::RecordOptions>();` work.